### PR TITLE
[Mellanox] add new platform msn2201

### PR DESCRIPTION
**Pull request by tomer-israel**
Signed-off-by: tomeri <tomeri@nvidia.com>

#### Why I did it
add new files for the new platform MSN2201

#### How I did it
added new device folder and platform device files to it based on (msn2700)
some of the files are linked to msn2700 and some were changed according to the new device specification (ports, speeds)
#### How to verify it
run with these files on the new HW
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [x] 202106

#### A picture of a cute animal (not mandatory but encouraged)

### CODE REVIEW

**Comment by dprital** [comment]
Please remove the backport from 202012.

File: device/mellanox/x86_64-mlnx_msn2201-r0/sensors.conf [hunk 1]
```diff
@@ -0,0 +1,81 @@
+################################################################################
```
Comment (dprital) [inline device/mellanox/x86_64-mlnx_msn2201-r0/sensors.conf]: Please change the Copyright header from Mellanox to NVIDIA header

**Review by dprital — COMMENTED** [review]
Please add NVIDIA Copyright header for the files

**Comment by keboliu** [comment]
1.	Currently all other existing platforms are all installed with SFP cages,  this is the first time we have RJ45, so in platform API chassis object initialization, it should ignore the RJ45 ports. You may need to start from here to check https://github.com/Azure/sonic-buildimage/blob/master/platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py#L68
2.	You need to fill in device data for 2201 in https://github.com/Azure/sonic-buildimage/blob/master/platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py

**Review by dprital — APPROVED** [review]

**Comment by moshemos** [comment]
1.	Buffer configuration. Should be approved by @Stephen Sun - **Done! should be as SN2700**
2.	How can pcied is provided? Who provided the info on the new system? was Yehuda and his team confirmed it? this is no Nvidia switch but rather a DNI switch so it might be the same and might not.
3.	Is the default speed of the ports is 1G and not 100M? is this aligned with the MRD? 
4.	SAI profile was approved by Itai? No - Sai will be supported 2 days after Itay will get the platform for bring up
5.	device/mellanox/x86_64-mlnx_msn2201-r0/platform.json: 
  a.	Product name is 2700. Wrong. 
  b.	Are there 3 CPLD files for this system? who confirmed?
  c.	Fans configuration – who confirmed? It was not tested to real hw 
  d.	Psu configuration – who confirmed? It was not tested to real hw
  e.	Thermal configuration – who confirmed? It was not tested to real hw
  f.	1G/100M should be supported. Missing 100m support for all ports. Not aligned with the mrd. 
6.	device/mellanox/x86_64-mlnx_msn2201-r0/platform_components.json 
There is an assumption we are upble to upgrade them. Was this discussed with Yehuda? Are the tools used today are good enough? there is open item on the alligator weekly which is not solved yet.
7.	device/mellanox/x86_64-mlnx_msn2201-r0/sensors.conf
a.	Device name is 2700. Wrong. 
b.	As there is no real hw how these file is generated?

**Comment by tomer-israel** [comment]
> 1. Buffer configuration. Should be approved by @stephen Sun - **Done! should be as SN2700**
> 2. How can pcied is provided? Who provided the info on the new system? was Yehuda and his team confirmed it? this is no Nvidia switch but rather a DNI switch so it might be the same and might not.
> 3. Is the default speed of the ports is 1G and not 100M? is this aligned with the MRD?
> 4. SAI profile was approved by Itai? No - Sai will be supported 2 days after Itay will get the platform for bring up
> 5. device/mellanox/x86_64-mlnx_msn2201-r0/platform.json:
>    a.	Product name is 2700. Wrong.
>    b.	Are there 3 CPLD files for this system? who confirmed?
>    c.	Fans configuration – who confirmed? It was not tested to real hw
>    d.	Psu configuration – who confirmed? It was not tested to real hw
>    e.	Thermal configuration – who confirmed? It was not tested to real hw
>    f.	1G/100M should be supported. Missing 100m support for all ports. Not aligned with the mrd.
> 6. device/mellanox/x86_64-mlnx_msn2201-r0/platform_components.json
>    There is an assumption we are upble to upgrade them. Was this discussed with Yehuda? Are the tools used today are good enough? there is open item on the alligator weekly which is not solved yet.
> 7. device/mellanox/x86_64-mlnx_msn2201-r0/sensors.conf
>    a.	Device name is 2700. Wrong.
>    b.	As there is no real hw how these file is generated?

1. done
2. Yehuda's team gave us the values of the pcie
3. the default speed is 1G, according to Haim Raz: "Always max, 1G for the RJ45 base-t and 100G for the qsfp-28."
4. on the sai xml the "port-speed" for the RJ-45 is unknown, itai needs to supply the port-speed for the rj45
5. i discussed with the hw-management team about the values: cpld, fans, psu, thermal values should be fine.
about the 1g/100m speeds on dynamic port breakout: I didnt add the 100M for the DPB , I have a different PR for this support and we need to think if the 100M is needed for the DPB mode.
https://github.com/tomer-israel/sonic-buildimage/pull/9
6. I will take it with Yehuda
7. new file was received from hw-management

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py [hunk 2]
```diff
@@ -334,6 +334,21 @@
                 THERMAL_DEV_FAN_AMBIENT
             ]
         )
+    },
+    # 13 2201
+    {
+        THERMAL_DEV_CATEGORY_CPU_CORE:(0, 2),
+        THERMAL_DEV_CATEGORY_MODULE:(1, 56),
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py]: Please check with Vadim/Michael that we can use all the ports for thermal. Not sure that the ports which are RJ45 are. maybe we can specific a range of the 4 100g ports only?

File: platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py [hunk 3]
```diff
@@ -271,5 +271,26 @@
             'hot_swappable': True,
             'led_num': 1
         }
-    }
+    },
+    'x86_64-nvidia_sn2201-r0': {
+        'thermal': {
```
Comment (liat-grozovik) [inline platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py:276]: Please check with Vadim/Michael if this is the right definition for termal. I am not aware we got a clear statement.

File: platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py (hunk 2 above)
Comment (tomer-israel) [reply platform/mellanox/mlnx-platform-api/sonic_platform/thermal.py]: RJ45 don't have thermals.
I will change it to, so we need to change it to:
THERMAL_DEV_CATEGORY_MODULE:(52, 4)

I will test it and update it

File: platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py (hunk 3 above)
Comment (tomer-israel) [reply platform/mellanox/mlnx-platform-api/sonic_platform/device_data.py:276]: answer from vadim:
"Regarding “minimum_table” – as far as I know we didn’t get dynamic minimum table for Alligator yet. In hw-mgmt thermal algorithm we consider all corners to be at 60%."

**Comment by liorghub** [comment]
Code taken to:
https://github.com/liorghub/sonic-buildimage/pull/20

**Pull request closed without merging**